src/axom/primal/geometry/Plane.hpp

```diff
@@ -209,6 +214,17 @@ class Plane
   AXOM_HOST_DEVICE int getOrientation(const PointType& x,
                                       double TOL = 1.e-9) const;
 
+  /*!
+   * \brief Simple check for validity of a Plane
+   *
+   * Check that the normal is not the zero vector.
+   *
+   * \param [in] TOL user-supplied tolerance. Optional. Default is 1.0e-50.
+   *
+   * \return True, if the Plane is valid, False otherwise
+   */
+  AXOM_HOST_DEVICE bool isValid(T TOL = static_cast<T>(PRIMAL_TINY)) const;
```

Comment:
Should `T` be a `double`? 
We'd likely have a problem here if `T` were not floating point (`float` or `double`), and the tolerances in other functions, e.g. `getOrientation()`

Reply:
isNearlyEqual expects all arguments (including the tolerance) to be the same type. Should they all be doubles or should they all be type T in Plane<T, NDIMS>? If Plane held integers in the current implementation, the tolerance would be 0, which would actually make sense. Maybe I'm thinking about it incorrectly, though.

Reply:
What would you like me to do?

Reply:
I was thinking the signature could be:
```cpp
AXOM_HOST_DEVICE bool isValid(double TOL = PRIMAL_TINY) const;
```
and if necessary, if could be `static_cast<T>` when calling `isNearlyEqual`.

Reply:
OK thanks. Done.